src/main/java/net/dv8tion/jda/internal/requests/restaction/MessageEditActionImpl.java

```diff
@@ -0,0 +1,89 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.requests.restaction;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.requests.Request;
+import net.dv8tion.jda.api.requests.Response;
+import net.dv8tion.jda.api.requests.restaction.MessageEditAction;
+import net.dv8tion.jda.api.utils.AttachedFile;
+import net.dv8tion.jda.api.utils.FileUpload;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.api.utils.messages.MessageEditBuilder;
+import net.dv8tion.jda.api.utils.messages.MessageEditData;
+import net.dv8tion.jda.internal.requests.RestActionImpl;
+import net.dv8tion.jda.internal.requests.Route;
+import net.dv8tion.jda.internal.utils.message.MessageEditBuilderMixin;
+import okhttp3.RequestBody;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+public class MessageEditActionImpl extends RestActionImpl<Message> implements MessageEditAction, MessageEditBuilderMixin<MessageEditAction>
+{
+    private final MessageChannel channel;
+    private final MessageEditBuilder builder = new MessageEditBuilder();
+
+    public MessageEditActionImpl(MessageChannel channel, String messageId)
+    {
+        super(channel.getJDA(), Route.Messages.EDIT_MESSAGE.compile(channel.getId(), messageId));
+        this.channel = channel;
+    }
+
+    @Override
+    public MessageEditBuilder getBuilder()
+    {
+        return builder;
+    }
+
+    @Override
+    protected RequestBody finalizeData()
+    {
+        try (MessageEditData data = builder.build())
+        {
+            DataObject json = data.toData();
+
+            List<FileUpload> files = data.getFiles();
+            if (files.isEmpty())
+                return getRequestBody(json);
+
+            return AttachedFile.createMultipartBody(files, json).build();
+        }
+    }
+
+    @Override
+    protected void handleSuccess(Response response, Request<Message> request)
+    {
+        request.onSuccess(api.getEntityBuilder().createMessageWithChannel(response.getObject(), channel, false));
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditAction setCheck(BooleanSupplier checks)
+    {
+        return (MessageEditAction) super.setCheck(checks);
+    }
+
+    @Nonnull
+    @Override
+    public MessageEditAction deadline(long timestamp)
+    {
+        return (MessageEditAction) super.deadline(timestamp);
+    }
```

Comment:
Can we use Fluent here?

Reply:
No, it clashes with RestActionImpl

Reply:
Ah i see. We _are_ using fluent in the interface but it conflicts in the impl